lib/base/tlsutility.cpp

```diff
@@ -962,7 +962,7 @@ bool VerifyCertificate(const std::shared_ptr<X509> &caCertificate, const std::sh
 	}
 
 	X509_STORE_CTX *csc = X509_STORE_CTX_new();
-	X509_STORE_CTX_init(csc, store, certificate.get(), nullptr);
+	X509_STORE_CTX_init(csc, store, certificate.get(), chain);
```

Comment:
... which is used for validation if given.

Reply:
see this (rather old thread): https://openssl-users.openssl.narkive.com/ztW5hA6Y/x509-store-ctx-init-function

`chain` just is any collection of certificates which will be used to find a path between the provided target-certificate and the provided target-ca.

So in this case the target-certificate is the (leaf)-certificate of the icinga-agent, the target-ca of course to top-root-ca-cert of the icinga-master and `chain` actually the whole bundle that the icinga-agent sends on establishing the connection (which includes every certificate in its local `ca.crt` and `cert.pem`).